indent/ecrystal.vim

```diff
@@ -84,28 +84,19 @@ else
   endfunction
 endif
 
-" Does the given pattern match at the cursor's position?
-function s:MatchCursor(pattern)
-  return searchpos(a:pattern, 'cnz', line('.')) == [line('.'), col('.')]
-endfunction
-
 " Does the given pattern match at the given position?
-function s:MatchAt(lnum, col, pattern)
-  let pos = getcurpos()
-
-  try
-    call cursor(a:lnum, a:col)
-    let result = s:MatchCursor(a:pattern)
-  finally
-    call setpos('.', pos)
-  endtry
+function! s:MatchAt(lnum, col, pattern) abort
+  return match(strcharpart(getline(a:lnum), a:col - 1), a:pattern) == 0
```

Comment:
With this implementation, it discards string before `col`. It means that this cannot match to a line when the pattern contains a behind match like `\@<=`. For example, when calling `s:MatchAt(1, 2, 'a\@<=bc')` and the line is `abc`, it should match but doesn't.

Instead, please use `{start}` optional parameter of `match()` here.
```suggestion
  return match(getline(a:lnum), a:pattern, a:col - 1) == a:col
```

Reply:
You're right, I didn't consider that.